src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -562,6 +564,7 @@ void GcsServer::InitKVManager() {
             std::make_unique<InMemoryStoreClient>(main_service_)));
   }
 
+  usage_reporter_ = std::make_shared<GcsUsageReporter>(main_service_, instance);
```

Comment:
Why don't we just not use shared pointer here? The reason is this will be broken if `main_service_ ` is GC'ed, so I feel like it's better making lifetime more explicit than using shared pointer (which can live beyond main_service_ which is not a shared pointer)